JavaScript Animations

karlasamantha opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/javascript-tutorial/pt.javascript.info?pullRequest=136) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/javascript-tutorial/pt.javascript.info?pullRequest=136) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/javascript-tutorial/pt.javascript.info?pullRequest=136) it.</sub>

kiniggit commented on 7-animation/3-js-animation/article.md:
```diff
@@ -1,157 +1,156 @@
+## Utilizando setInterval
 
-An animation can be implemented as a sequence of frames -- usually small changes to HTML/CSS properties.
+A animação pode ser implmentada como uma sequência de frames (quadros) -- geralmente pequenas modificações nas propriedades de HTML/CSS.
```
minor typo: implmentada -> implementada

kiniggit commented on 7-animation/3-js-animation/article.md:
````diff
@@ -1,157 +1,156 @@
 
 ```js
-let start = Date.now(); // remember start time
+let start = Date.now() // preservar o tempo inicial
````
please dont change the code, keep the ';'

kiniggit commented on 7-animation/3-js-animation/article.md:
```diff
@@ -1,157 +1,156 @@
-let timer = setInterval(function() {
-  // how much time passed from the start?
-  let timePassed = Date.now() - start;
+let timer = setInterval(function () {
```
please dont change the code, keep the same spaces as in the original

kiniggit commented on 7-animation/3-js-animation/article.md:
```diff
@@ -1,157 +1,156 @@
-  let timePassed = Date.now() - start;
+let timer = setInterval(function () {
+  // quanto tempo passou desde o início?
+  let timePassed = Date.now() - start
```
same here, keep the ';'

kiniggit commented on 7-animation/3-js-animation/article.md:
```diff
@@ -1,157 +1,156 @@
-  if (timePassed >= 2000) {
-    clearInterval(timer); // finish the animation after 2 seconds
-    return;
+  if (tempoPetimePassedrcorrido >= 2000) {
```
please also dont translate the variables names

kiniggit commented:
Hi, Karla! Doing fine? Please don't translate or change the source code in the examples. Could you take a look into that? Like, you should keep the ';', spaces and variable names. Cheers!

karlasamantha commented:
> Hi, Karla! Doing fine? Please don't translate or change the source code in the examples. Could you take a look into that? Like, you should keep the ';', spaces and variable names. Cheers!

Hey, yeah. I'm addressing feedback now. My linter changed the code, fixing that :)

karlasamantha commented:
/done

odsantos commented:
Hi @karlasamantha would you please also resolve the conflicts, or kindly inform if unavailable to? Thank you.

odsantos commented:
Hi @karlasamantha , please I wonder if you could resolve the conflicts, or kindly inform if unavailable to? Thank you.

karlasamantha commented:
> Hi @karlasamantha , please I wonder if you could resolve the conflicts, or kindly inform if unavailable to? Thank you.

Hi @odsantos , sorry for the delay! I'll try to fix the conflicts today!

karlasamantha commented:
/done

karlasamantha commented:
/done

odsantos commented:
Hi @karlasamantha , thank you.

odsantos commented on 7-animation/3-js-animation/1-animate-ball/solution.md:
```diff
@@ -1,10 +1,12 @@
+Para quicar, podemos usar a propriedade CSS `top` e `position:absolute` para a bola dentro do campo com `position:relative`.
 
-The bottom coordinate of the field is `field.clientHeight`. The CSS `top` property refers to the upper edge of the ball. So it should go from `0` till `field.clientHeight - ball.clientHeight`, that's the final lowest position of the upper edge of the ball.
+A coordenada inferior do campo é `field.clientHeight`. Mas a propriedade `top` fornece coordenadas para o topo da bola, a posição da borda é `field.clientHeight - ball.clientHeight`.
```
```suggestion
A coordenada inferior do campo é `field.clientHeight`. Mas a propriedade `top` fornece coordenadas para o topo da bola. Então animamos a propriedade `top` a partir de `0` até `field.clientHeight - ball.clientHeight`,  isto é até à posição mais baixa do topo da bola.
```

odsantos commented on 7-animation/3-js-animation/1-animate-ball/solution.md:
```diff
@@ -1,10 +1,12 @@
+A coordenada inferior do campo é `field.clientHeight`. Mas a propriedade `top` fornece coordenadas para o topo da bola, a posição da borda é `field.clientHeight - ball.clientHeight`.
 
-To get the "bouncing" effect we can use the timing function `bounce` in `easeOut` mode.
+Então animamos a propriedade `top` a partir de `0` até `field.clientHeight - ball.clientHeight`.
```
```suggestion
Para obter o efeito de "quique", podemos usar a função de tempo `bounce` no modo `easeOut`.
```

odsantos commented on 7-animation/3-js-animation/1-animate-ball/solution.md:
```diff
@@ -1,10 +1,12 @@
+Então animamos a propriedade `top` a partir de `0` até `field.clientHeight - ball.clientHeight`.
 
-Here's the final code for the animation:
+Para obter o efeito de "quique", podemos usar a função de tempo `bounce` no modo `easeOut`
```
```suggestion
Aqui está o código final para a animação:
```

odsantos commented on 7-animation/3-js-animation/1-animate-ball/solution.md:
```diff
@@ -1,10 +1,12 @@
 
-Here's the final code for the animation:
+Para obter o efeito de "quique", podemos usar a função de tempo `bounce` no modo `easeOut`
+
```
```suggestion
```

odsantos commented on 7-animation/3-js-animation/1-animate-ball/solution.md:
```diff
@@ -1,10 +1,12 @@
-Here's the final code for the animation:
+Para obter o efeito de "quique", podemos usar a função de tempo `bounce` no modo `easeOut`
+
+Aqui está o código final para a animação:
```
```suggestion
```js
```

odsantos commented on 7-animation/3-js-animation/1-animate-ball/solution.md:
```diff
@@ -1,10 +1,12 @@
+Para obter o efeito de "quique", podemos usar a função de tempo `bounce` no modo `easeOut`
+
+Aqui está o código final para a animação:
 
```
```suggestion
let to = field.clientHeight - ball.clientHeight;
```

odsantos commented on 7-animation/3-js-animation/1-animate-ball/solution.md:
````diff
@@ -1,10 +1,12 @@
+
+Aqui está o código final para a animação:
 
 ```js
````
```suggestion

```

odsantos commented on 7-animation/3-js-animation/1-animate-ball/solution.md:
````diff
@@ -1,10 +1,12 @@
+Aqui está o código final para a animação:
 
 ```js
 let to = field.clientHeight - ball.clientHeight;
````
````suggestion
animate({
  duration: 2000,
  timing: makeEaseOut(bounce),
  draw(progress) {
    ball.style.top = to * progress + 'px'
  }
});
```
````

odsantos commented on 7-animation/3-js-animation/2-animate-ball-hops/solution.md:
```diff
@@ -1,31 +1,31 @@
+Podemos escrever mais um `animate` para isso.
 
-As the time function we could use `linear`, but something like `makeEaseOut(quad)` looks much better.
+Como função de tempo podemos usar `liner`, mas algo como `makeEaseOut(quad)` parece bem melhor.
```
```suggestion
Como função de tempo podemos usar `linear`, mas algo como `makeEaseOut(quad)` parece bem melhor.
```

odsantos commented on 7-animation/3-js-animation/2-animate-ball-hops/solution.md:
````diff
@@ -1,31 +1,31 @@
 ```js
-let height = field.clientHeight - ball.clientHeight;
-let width = 100;
+let height = field.clientHeight - ball.clientHeight
````
```suggestion
let height = field.clientHeight - ball.clientHeight;
```

odsantos commented on 7-animation/3-js-animation/2-animate-ball-hops/solution.md:
```diff
@@ -1,31 +1,31 @@
-let height = field.clientHeight - ball.clientHeight;
-let width = 100;
+let height = field.clientHeight - ball.clientHeight
+let width = 100
```
```suggestion
let width = 100;
```

odsantos commented on 7-animation/3-js-animation/2-animate-ball-hops/solution.md:
```diff
@@ -1,31 +1,31 @@
+let height = field.clientHeight - ball.clientHeight
+let width = 100
 
 // animate top (bouncing)
```
```suggestion
// anime o topo (quicando)
```

odsantos commented on 7-animation/3-js-animation/2-animate-ball-hops/solution.md:
```diff
@@ -1,31 +1,31 @@
   }
 });
 
 // animate left (moving to the right)
```
```suggestion
// anime a esquerda (movendo para a direita)
```

odsantos commented on 7-animation/3-js-animation/2-animate-ball-hops/task.md:
```diff
@@ -2,12 +2,12 @@ importance: 5
 [iframe height=250 src="solution"]
 
-Write the animation code. The distance to the left is `100px`.
+Escreva o código de animação. A distância para a esquerda é `100px`.
```
```suggestion
Escreva o código da animação. A distância para a esquerda é `100px`.
```

odsantos commented on 7-animation/3-js-animation/2-animate-ball-hops/task.md:
```diff
@@ -2,12 +2,12 @@ importance: 5
+Escreva o código de animação. A distância para a esquerda é `100px`.
 
-Take the solution of the previous task <info:task/animate-ball> as the source.
+Use a solução da atividade anterior <info:task/animate-ball> como a fonte.
```
```suggestion
Use a solução da atividade anterior <info:task/animate-ball> como fonte.
```

odsantos commented on 7-animation/3-js-animation/text.view/index.html:
```diff
@@ -8,12 +8,11 @@
-And stood awhile in thought.
+  <textarea id="textExample" rows="5" cols="60">
+    Ele pegou sua espada vorpal na mão:
+    Faz tempo que o manxome inimigo que ele procurava -
```
```suggestion
    Faz tempo que o inimigo manxome ele procurava -
```

odsantos commented on 7-animation/3-js-animation/text.view/index.html:
```diff
@@ -8,12 +8,11 @@
+    E ficou um pouco em pensamento.
   </textarea>
 
   <button onclick="animateText(textExample)">Run the animated typing!</button>
```
```suggestion
  <button onclick="animateText(textExample)">Corra o texto animado!</button>
```

odsantos reviewed (changes requested):
Hi @karlasamantha ,

The article review is still missing, hopefully to another time.
Please, feel free to leave a note on something you may not agree.

Thank you.

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -199,87 +199,87 @@ animate({
 [iframe height=40 src="quad" link]
 
-...Or the cubic curve or even greater `n`. Increasing the power makes it speed up faster.
+...Ou a curva cúbica ou evento maior que `n`. Aumentar a potência faz com que acelere mais rapidamente.
```
```suggestion
...Ou a curva cúbica ou até um `n` maior. Aumentar a potência faz com que acelere mais rapidamente.
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -291,93 +291,93 @@ function bounce(timeFraction) {
 [iframe height=40 src="elastic" link]
 
-## Reversal: ease*
+## Reversão: ease\*
```
```suggestion
## Reversão: ease*
```

odsantos commented on 7-animation/3-js-animation/article.md:
````diff
@@ -291,93 +291,93 @@ function bounce(timeFraction) {
 ```
 
-In other words, we have a "transform" function `makeEaseOut` that takes a "regular" timing function and returns the wrapper around it:
+Em outras palavras, temos uma função de "transformar" `makeEaseOut` que recebe a função de tempo "regular" e retorna a função que a envolve:
````
```suggestion
Em outras palavras, temos uma função "transformar" `makeEaseOut` que recebe a função de tempo "regular" e retorna a função que a envolve:
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -291,93 +291,93 @@ function bounce(timeFraction) {
-- Regular bounce -- the object bounces at the bottom, then at the end sharply jumps to the top.
-- After `easeOut` -- it first jumps to the top, then bounces there.
+- Quique regular -- o objeto salta na parte inferior, e no final pula abruptamente para o topo.
+- Depois `easeOut` -- ele pula no começo, depois quica naquele lugar.
```
```suggestion
- Depois de `easeOut` -- ele primeiro pula para o topo, e depois quica naquele lugar.
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -291,93 +291,93 @@ function bounce(timeFraction) {
+Podemos também mostrar o efeito tanto no começo quanto no final da animação. A transformação é chamada "easeInOut".
 
-Given the timing function, we calculate the animation state like this:
+Dado a função de tempo, calculamos o estado de animação assim:
```
```suggestion
Dada a função de tempo, calculamos o estado de animação assim:
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -387,69 +387,68 @@ function makeEaseInOut(timing) {
+A transformaçào "easeInOut" une dois gráficos em um: `easeIn` (regular) para a primeira metade da animação e `easeOut` (invertida) -- para a segunda parte.
 
-The effect is clearly seen if we compare the graphs of `easeIn`, `easeOut` and `easeInOut` of the `circ` timing function:
+O efeito é claramente visto se compararmos os gráficos de `easeIn`, `easeOut` e `easeInOut` da função `circ`:
```
```suggestion
O efeito é claramente visto se compararmos os gráficos de `easeIn`, `easeOut` e `easeInOut` da função de tempo `circ`:
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -387,69 +387,68 @@ function makeEaseInOut(timing) {
-- <span style="color:#8DB173">Green</span> -- `easeOut`.
+- <span style="color:#EE6B47">Red</span> é a variante regular de `circ` (`easeIn`).
+- <span style="color:#8DB173">Greed</span> -- `easeOut`.
 - <span style="color:#62C0DC">Blue</span> -- `easeInOut`.
```
```suggestion
- <span style="color:#62C0DC">Azul</span> -- `easeInOut`.
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -387,69 +387,68 @@ function makeEaseInOut(timing) {
+Ao invés de mover o elemento, podemos fazer outra coisa. Tudo o que precisamos é escrever a `draw` apropriada.
 
-Here's the animated "bouncing" text typing:
+Aqui está o texto animato "quicando":
```
```suggestion
Aqui está o texto animado "quicando":
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -387,69 +387,68 @@ function makeEaseInOut(timing) {
+## Resumo
 
-For animations that CSS can't handle well, or those that need tight control, JavaScript can help. JavaScript animations should be implemented via `requestAnimationFrame`. That built-in method allows to setup a callback function to run when the browser will be preparing a repaint. Usually that's very soon, but the exact time depends on the browser.
+Para animação que CSS não lida bem, ou aquelas que precisam de controle rígido, JavaScript pode ajudar. Animações JavaScript devem ser implementadas via `requestAnimationFrame`. Esse método embutido permite configurar uma callback para que seja executada quando o navegador estiver preparando uma repintura. Geralmente é bem breve, mas o tempo exato depende do navegador.
```
```suggestion
Para animação que CSS não lida bem, ou aquelas que precisam de controle rígido, JavaScript pode ajudar. Animações com JavaScript devem ser implementadas via `requestAnimationFrame`. Esse método embutido permite configurar uma callback para que seja executada quando o navegador estiver preparando uma repintura. Geralmente é bem breve, mas o tempo exato depende do navegador.
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -387,69 +387,68 @@ function makeEaseInOut(timing) {
+Quando uma página está em segundo plano, não ocorrem repinturas, então a callback não será executada: a animação será suspensa e não consumirá recursos. Isso é ótimo.
 
-Here's the helper `animate` function to setup most animations:
+Aqui está a função helper `animate` para configura a maioria das animações:
```
```suggestion
Aqui está uma função de ajuda `animate` para configurar a maioria das animações:
```

odsantos commented on 7-animation/3-js-animation/article.md:
```diff
@@ -387,69 +387,68 @@ function makeEaseInOut(timing) {
+Certamente poderíamos melhorá-las, adicionar mais recursos adicionais supérfluos, mas animações JavaScript não são aplicadas no dia-a-dia. Elas são usadas para fazer coisas mais interessantes e não-padrão. Então você gostaria de adicionar novas funcionalidades quando precisa delas.
 
-JavaScript animations can use any timing function. We covered a lot of examples and transformations to make them even more versatile. Unlike CSS, we are not limited to Bezier curves here.
+Animações JavaScript podem usar qualquer função de tempo. Cobrimos muitos exemplos e transformações para fazê-las ainda mais versáteis. Ao contrário de CSS, não estamos limitados às curvas de Bezier.
```
```suggestion
Animações com JavaScript podem usar qualquer função de tempo. Cobrimos muitos exemplos e transformações para fazê-las ainda mais versáteis. Ao contrário de CSS, não estamos limitados às curvas de Bezier.
```

karlasamantha commented:
/done

nazarepiedady commented:
@odsantos, @jonnathan-ls, @Peruibeloko, what do you recommend colleagues to be done in this pull request?

nazarepiedady commented:
@karlasamantha, are you available to update and apply the changes requested?

karlasamantha commented:
> @karlasamantha, are you available to update and apply the changes requested?

@nazarepiedady sure, but I don't see any pending changes requested. Am I missing something here?

nazarepiedady commented:
@karlasamantha, I would like to invite you to compare the original English content to the Portuguese content translated by you to see if there are any updates needed or sentences that need to be applied.

I also invite you to check if the reviews did by @odsantos are observed or resolved.

nazarepiedady commented:
@jonnathan-ls, do you have time to review and handle this pull request?

nazarepiedady commented:
@Peruibeloko, do you have some time to deal with this pull request?

gabifs commented:
Hey @nazarepiedady , this PR still need one more approve
